Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -860,33 +860,77 @@ WebIDL::ExceptionOr<GC::Ref<PendingResponse>> scheme_fetch(JS::Realm& realm, Inf
             auto content_type_header = Infrastructure::Header::from_string_pair("Content-Type"sv, type);
             response->header_list()->append(move(content_type_header));
         }
-        // FIXME: 9. Otherwise:
+        // 9. Otherwise:
         else {
             // 1. Set response’s range-requested flag.
+            response->set_range_requested(true);
+
             // 2. Let rangeHeader be the result of getting `Range` from request’s header list.
+            auto const range_header = request->header_list()->get("Range"sv.bytes()).value_or(ByteBuffer {});
+
             // 3. Let rangeValue be the result of parsing a single range header value given rangeHeader and true.
+            auto maybe_range_value = Infrastructure::parse_single_range_header_value(range_header, true);
+
             // 4. If rangeValue is failure, then return a network error.
+            if (!maybe_range_value.has_value())
+                return PendingResponse::create(vm, request, Infrastructure::Response::network_error(vm, "Failed to parse single range header value"sv));
+
             // 5. Let (rangeStart, rangeEnd) be rangeValue.
+            auto& [range_start, range_end] = maybe_range_value.value();
+
             // 6. If rangeStart is null:
-            //     1. Set rangeStart to fullLength − rangeEnd.
-            //     2. Set rangeEnd to rangeStart + rangeEnd − 1.
+            if (!range_start.has_value()) {
+                //     1. Set rangeStart to fullLength − rangeEnd.
```

Comment:
Spec steps don't need to be indented within if/else blocks.

Also, it might be worth adding a `VERIFY(range_end.has_value())` here, to make it clear that by this point we expect `range_end` must have a value.

Reply:
Done.